tests/unit/test_utils.py

```diff
@@ -1263,3 +1264,26 @@ def test_deprecated_message_reads_well():
         "You can find discussion regarding this at "
         "https://github.com/pypa/pip/issues/100000."
     )
+
+
+@pytest.mark.parametrize("setup_py_path", [
+    "setup.py",
+    "/dir/path/setup.py",
+    "../relative/path/setup.py",
```

Comment:
These additional path forms don't really add to the test. I would just pick one (e.g. the second) and use it in the test (and hard-code your `args[-1]` assertions rather than using a format string).